src/cron.rs

```diff
@@ -134,51 +139,63 @@ pub fn update_nodes(
     let changed : Vec<(String, NodeData)> = db.transaction::<_, anyhow::Error, _>(|| {
         let mut changed = Vec::new();
 
-        // Go over every node in the database
-        let db_nodes = nodes::table.load::<models::NodeQuery>(db)?;
-        for db_node in db_nodes.into_iter() {
-            let (id, db_data) = model_to_node_data(db_node);
-            if let Some(cur_data) = cur_nodes_map.remove(&id) {
-                // We already know this node.
-                // Did it change?
-                if cur_data != db_data {
-                    // Update in database
-                    diesel::update(nodes::table.find(id.as_str()))
-                        .set((
-                            nodes::name.eq(cur_data.name.as_str()),
-                            nodes::online.eq(cur_data.online)
-                        ))
+        // Go over all nodes in the hash map to count how many are online
+        for (_id, cur_data) in cur_nodes_map.into_iter() {
+            if cur_data.online {
+                count_online_nodes += 1;
+            }
+        }
+
+        if count_online_nodes < config.ui.min_online_nodes_on_map {
+            // don't do anything if nearly all nodes are offline
+            bail!(NodeListError::MapOnlineStatusChangedToOffline { num_nodes: count_online_nodes });
```

Comment:
`bail!` is for error conditions, and will result in a 503 response. That does not seem like the appropriate reaction here. Instead the page should likely return something like

> Skipped sending notifications as fewer than X nodes are online

The cronjob will send print that (via `curl`) and send a notification email to the admin.